clients/callback_client.go

```diff
@@ -86,21 +90,54 @@ func (c CallbackClient) SendTranscodeStatusError(callbackURL, errorMsg string) e
 	return c.DoWithRetries(r)
 }
 
+// Calculate the overall completion ratio based on the completion ratio of the current stage.
+// The weighting will need to be tweaked as we understand better the relative time spent in the
+// segmenting vs. transcoding stages.
+func overallCompletionRatio(status TranscodeStatus, currentStageCompletionRatio float64) float64 {
+	// Sanity check the inputs are within the 0-1 bounds
+	if currentStageCompletionRatio < 0 {
+		currentStageCompletionRatio = 0
+	}
+	if currentStageCompletionRatio > 1 {
+		currentStageCompletionRatio = 1
+	}
+
+	// Define the "base" numbers - e.g the overall ratio we start each stage at
+	var TranscodeStatusPreparingBase float64 = 0.0
+	var TranscodeStatusPreparingTranscodingBase float64 = 0.4
+	var TranscodeStatusCompletedBase float64 = 1
+
+	switch status {
+	case TranscodeStatusPreparing:
+		return TranscodeStatusPreparingBase + (currentStageCompletionRatio * (TranscodeStatusPreparingTranscodingBase - TranscodeStatusPreparingBase))
+	case TranscodeStatusPreparingCompleted:
+		return TranscodeStatusPreparingTranscodingBase
+	case TranscodeStatusTranscoding:
+		return TranscodeStatusPreparingTranscodingBase + (currentStageCompletionRatio * (TranscodeStatusCompletedBase - TranscodeStatusPreparingTranscodingBase))
```

Comment:
Another way of implementing this, to have the more complex math only in 1 place, would be to have 2 vars "start" and "end" and force the `currentStageCompletionRatio` to be `1` on the "Completed" steps (but use their same start/end as the non-completed). Sth like
```go
	switch status {
	case TranscodeStatusPreparing, TranscodeStatusPreparingCompleted:
		start, end = 0, 0.4
	case TranscodeStatusTranscoding, TranscodeStatusTranscodingCompleted:
		start, end = 0.4, 1
        }
        if (status == TranscodeStatusPreparingCompleted || status == TranscodeStatusTranscodingCompleted) {
                currentStageCompletionRatio = 1
        }
```


Then there could be a single progress-scaling function like this: https://github.com/livepeer/task-runner/blob/4d1ae1768c01e979969bf7f00b81267db05e299c/task/progress.go#L66

Certainly not a required change, just tripping on the impl. Could make sense when we start adding more steps here. WDYT?

Reply:
I like that a lot!